File: Programmers/Lv1/문자열 나누기.py
def solution(s):
    answer = 0
    dict = {}
    for w in s:
        if len(dict.keys()) > 0:
            tmp = list(dict.keys())[0]
            if w == tmp:
                dict[w] += 1
            else:
                dict['else'] = dict.get('else', 0) + 1
        else:
            dict[w] = 1

        if len(dict.keys()) == 2 and len(set(dict.values())) == 1:
            answer += 1
            dict = {}

    if len(dict.keys()) != 0:
        answer += 1
    return answer

File: Programmers/Lv1/문자열 나누기.js
function solution(s) {
    var res = [];

    var now = 0;
    var val = 0;
    var otherVal = 0;
    
    flag = false;
    for(let i = 0; i<s.length;i++){
        if(s[now] == s[i]){
            val += 1;        
        } else {
            otherVal += 1;
        }
        
        if (val == otherVal){
            flag = true;
            if (flag){
                res.push(s.slice(now, i + 1));
                now = i + 1;
                flag = false;
                val = 0;
                otherVal = 0;
            }
        }
    }
    
    if(now < s.length && val !== otherVal){
        res.push(s.slice(now))
    }
    
    
    return res.length;
}